syntax = "proto3";

service Gateway {
    rpc ligarLampada(Empty) returns (Reply);
    rpc desligarLampada(Empty) returns (Reply);

    rpc ligarAr(Empty) returns (Reply);
    rpc desligarAr(Empty) returns (Reply);

    rpc ligarSom(Empty) returns (Reply);
    rpc desligarSom(Empty) returns (Reply);

    rpc aumentarTemperatura(Empty) returns (Reply);
    rpc diminuirTemperatura(Empty) returns (Reply);

    rpc aumentarVolume(Empty) returns (Reply);
    rpc diminuirVolume(Empty) returns (Reply);
}
//Não precisa enviar nada na hora da requisição
message Empty {}

//Mensagem de resposta dos atuadores para o Home Assistant
message Reply {
    string response = 1;
}
/* Estruturação alternativa>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
service Gateway {
    rpc commando(Command) returns (Reply);
}

message Command {
    Type type = 1 [default = LAMPADA];
    Operation operation = 2 [default = LIGAR];

    enum Type {
        LAMPADA = 0;
        ARCONDICIONADO = 1;
        SISTEMADESOM = 2;
    }

    enum Operation {
        LIGAR = 0;
        DESLIGAR = 1;
        AUMENTAR = 2;
        DIMINUIR = 3;
}

message Reply {
    string response = 1;
}
*/
